File: src/go/credit-api/modules/credit/credit.go
package credit

import "github.com/google/uuid"

type Credit struct {
	Id           string        `json:"id"`
	Name         string        `json:"name"`
	Transactions []Transaction `json:"transactions"`

	etag            string
	newTransactions []Transaction
}

type Transaction struct {
	Id    string `json:"id"`
	Value int    `json:"value"`
}

func (c *Credit) AddTransaction(value int) {
	transaction := Transaction{Id: uuid.New().String(), Value: value}
	c.Transactions = append(c.Transactions, transaction)
	c.newTransactions = append(c.newTransactions, transaction)
}

func (c *Credit) GetNewTransactions() (transactions []Transaction) {
	return c.newTransactions
}

func (c *Credit) HasNewTransactions() bool {
	return len(c.newTransactions) > 0
}

func (c *Credit) SetEtag(etag string) {
	c.etag = etag
}

func (c *Credit) GetEtag() string {
	return c.etag
}

File: src/dotnet/credit-api/Modules/Credit/Credit.cs
using System.Text.Json.Serialization;

namespace CreditApi.Modules.Credit;

public class Credit
{
    public string Id { get; init; }
    public string Name { get; init; }
    public DateOnly CurrentMonth { get; set; }
    public ICollection<Transaction> Transactions { get; init; } = new List<Transaction>();
    [JsonIgnore]
    public IReadOnlyCollection<Transaction> NewTransactions => _newTransactions;

    private readonly List<Transaction> _newTransactions = new();

    public void AddTransaction(int value, DateOnly transactionDate)
    {
        var newTransaction = new Transaction { Id = Guid.NewGuid().ToString(), Value = value, TransactionDate = transactionDate};
        Transactions.Add(newTransaction);
        _newTransactions.Add(newTransaction);
    }

    public int CloseCurrentMonth()
    {
        var month = CurrentMonth.Month;
        CurrentMonth = CurrentMonth.AddMonths(1);
        return month;
    }
}

public class Transaction
{
    public required string Id { get; init; }
    public required int Value { get; init; }
    public required DateOnly TransactionDate { get; init; }
}